File: vdp-php/tests/mlc/CardholderEnrollmentTest.php
<?php

namespace Vdp;

class CardholderEnrollmentTest extends \PHPUnit_Framework_TestCase {
	
	public function setUp() {
		$this->conf = parse_ini_file ( "configuration.ini", true );
		$this->abstractVisaAPIClient = new AbstractVisaAPIClient;
		$this->enrollementData = json_encode ([
		  "enrollmentMessageType" => "enroll",
		  "enrollmentRequest" => [
		    "cardholderMobileNumber" => "0016504323000",
		    "clientMessageID" => $this->conf ['VDP'] ['mlcClientMessageID'],
		    "deviceId" => $this->conf ['VDP'] ['mlcDeviceId'],
		    "issuerId" => $this->conf ['VDP'] ['mlcIssuerId'],
		    "primaryAccountNumber" => $this->conf ['VDP'] ['mlcPrimaryAccountNumber']
		  ]
		]);
	}
	
	public function testCardEnrollment() {
		$baseUrl = "mlc/";
		$resourcePath = "enrollment/v1/enrollments";
		$statusCode = $this->abstractVisaAPIClient->doMutualAuthCall ( 'post', $baseUrl.$resourcePath, 'MLC Card Enrollement Test', $this->enrollementData );
		$this->assertEquals($statusCode, "200");
	}
}

File: vdp-ruby/test/mlc/cardholder_enrollment_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/abstract_visa_api_client', __FILE__)

class CardholderEnrollmentTest < Test::Unit::TestCase
  def setup
    @abstract_visa_api_client = AbstractVisaAPIClient.new
    @config = YAML.load_file('configuration.yml')
    @enrollementData ='''{
    "enrollmentMessageType": "enroll",
    "enrollmentRequest": {
      "cardholderMobileNumber": "0016504323000",
      "clientMessageID": "'''+ @config['mlcClientMessageID'] + '''",
      "deviceId": "'''+ @config['mlcDeviceId'] + '''",
      "issuerId": "'''+ @config['mlcIssuerId'] + '''",
      "primaryAccountNumber": "'''+ @config['mlcPrimaryAccountNumber'] + '''"
    }
  }'''
  end
  
  def test_cardEnrollment
    base_uri = 'mlc/'
    resource_path = 'enrollment/v1/enrollments'
    response_code = @abstract_visa_api_client.doMutualAuthRequest("#{base_uri}#{resource_path}", "Cardholder Enrollment test", "post", @enrollementData)
    assert_equal("200", response_code, "Cardholder Enrollment test failed")
  end
end